Fig.Tests/TypedSettingsTests.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace Fig.Test
+{
+    public class TypedSettingsTests
+    {
+        private Settings _settings;
+
+        [SetUp]
+        public void Setup()
+        {
+            this._settings = new SettingsBuilder()
+                .UseSettingsDictionary(new SettingsDictionary() {
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Name)}"] = "Fullname",
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Age)}"] = "40",
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Pi)}"] = "3.14"
```

Comment:
Nice use of nameof() here, we should use that in my messy tests too

Reply:
Makes refactoring easy, since there aren't any magic strings.  One downside is that it's kinda verbose.